core/src/main/scala/org/apache/spark/scheduler/SchedulerResourceInformation.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.annotation.Evolving
+
+/**
+ * Class to hold information about a type of Resource used by the scheduler. This is a separate
+ * class from the ResourceInformation because here its mutable because the scheduler has to update
+ * the addresses based on what its assigned and what is available.
+ */
+@Evolving
+private[spark] class SchedulerResourceInformation(
```

Comment:
The class name is a little confusing. It is not `SchedulerResource`. How about `ExecutorResourceAllocationInfo`?

Reply:
updated